sync/message/payload/aleyk.go

```diff
@@ -5,19 +5,20 @@ import (
 
 	"github.com/libp2p/go-libp2p-core/peer"
 	"github.com/zarbchain/zarb-go/crypto"
+	"github.com/zarbchain/zarb-go/crypto/bls"
 	"github.com/zarbchain/zarb-go/errors"
 	"github.com/zarbchain/zarb-go/version"
 )
 
 type AleykPayload struct {
-	ResponseTarget  peer.ID          `cbor:"1,keyasint"`
-	ResponseCode    ResponseCode     `cbor:"2,keyasint"`
-	ResponseMessage string           `cbor:"3,keyasint"`
-	NodeVersion     string           `cbor:"4,keyasint"`
-	Moniker         string           `cbor:"5,keyasint"`
-	PublicKey       crypto.PublicKey `cbor:"6,keyasint"`
-	Height          int              `cbor:"7,keyasint"`
-	Flags           int              `cbor:"8,keyasint"`
+	ResponseTarget  peer.ID        `cbor:"1,keyasint"`
+	ResponseCode    ResponseCode   `cbor:"2,keyasint"`
+	ResponseMessage string         `cbor:"3,keyasint"`
+	NodeVersion     string         `cbor:"4,keyasint"`
+	Moniker         string         `cbor:"5,keyasint"`
+	PublicKey       *bls.PublicKey `cbor:"6,keyasint"`
```

Comment:
this way we are tightly coupled to bls implementation of public key while I think you are trying to decouple it

Reply:
The problem here is we can't deserialize public Key interface. To deserialized public key we have two choices:
- Don't use interface (current approach)
- Adding type to the the struct and based of type we deserialize the public key.

Reply:
if we can't de/serialize the public key what are marshal functionalities for?
I think we should have and identifier to indicate the type of public key rather than directly using implementer(bls)
```
type PublicKey interface {
	RawBytes() []byte
	String() string
	MarshalJSON() ([]byte, error)
	MarshalCBOR() ([]byte, error)
	SanityCheck() error
	Verify(msg []byte, sig Signature) bool
	Address() Address
	EqualsTo(right PublicKey) bool
}
```